src/Components/Capture.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include "Component.h"
+#include <opencv2/opencv.hpp>
+
+class Render;
+class Encoder;
+
+class Capture: public Component<int> {
+public:
+	Capture(Render*, Encoder*);
+protected:
+	void start();
+	bool wait();
+	void process();
+	void end();
+private:
+	Render* render;
+	Encoder* encoder;
+	cv::VideoCapture videoCapture;
+	string name;
```

Comment:
If "name" is common to all subclass of Component, consider moving it into Component.